README.md

```diff
@@ -1 +1,76 @@
-# angular-what-is-a-filter-readme
\ No newline at end of file
+# What is a Filter?
+
+## Overview
+
+Angular offers us the powerful concept of filters. Let's take a look at what they can offer us.
+
+## Objectives
+
+- Show an Angular filter
+- Use a filter on an ng-repeat
+- Use a filter on an input's expressed value
+
+## Filtering our datasets
+
+Filters are simple functions that we can use to manipulate (or filter) data in Angular.
+
+We've touched on the `ng-repeat` directive for repeating a list of data - wouldn't it be neat if we could filter on that data too? This comes in useful when we a large dataset and would like to quickly search through it to find the relevant data.
+
+We can do this using the pipe operator (`|`) after the ng-repeat. The syntax for this is `| filterName: value`. We can use different filters (by changing the `fieldName` part) and change what the filter criteria is by changing the `value` part.
+
+The filter that we'd use to filter an ng-repeat by a value is simply named `filter`! We can pass a variable to this filter and it will search the whole dataset for us.
+
+```html
+<input ng-model="$ctrl.search" />
+
+<ul>
+	<li ng-repeat="data in $ctrl.data | filter: $ctrl.search">
+	</li>
+</ul>
+```
+
+This will filter our dataset by the value of the input data - providing us with a really simple yet powerful search filter!
```

Comment:
Break this down a little more with a real context. Explain that as the input gets updated, the displayed results will get updated automatically. If our list is a list of names and we type "Bob" into the input box, only names containing "Bob" will be displayed.